### ISSUE TITLE
Early stopping based on validation accuracy

### ISSUE THREAD

**Issue by 18-RAJAT** [18-RAJAT/Interactive-Production-text2sql-Pipeline#13]
Implement early stopping using validation exact-match accuracy instead of just training loss to prevent overfitting.
